packages/react-components/src/WithWalletConnector.ts

```diff
@@ -65,26 +105,40 @@ interface State {
     /**
      * The currently active connection.
      *
-     * For reasons explained in {@link activeConnector},
-     * there's no guarantee that this connection originates from the current value of {@link activeConnector}
-     * as there's no automatic synchronization between them.
+     * As explained in {@link activeConnector},
+     * this connection generally might not originate from {@link activeConnector}.
      */
     activeConnection: WalletConnection | undefined;
 
     /**
-     * The hash of the genesis block for the chain that {@link activeConnectedAccount} lives on.
+     * A map from open connections to their selected accounts.
+     * Connections without a selected account will not have an entry in this map.
+     */
+    connectedAccounts: Map<WalletConnection, string>;
+
+    /**
+     * A map from open connections to the hash of the genesis block for the chain that the selected accounts
+     * of the connections live on.
+     * Connections without a selected account (or the account's chain is unknown) will not have an entry in this map.
      *
-     * TODO The value of this field is not too reliable as it's updated only when the `onChainChanged` event fires.
+     * TODO The reported hash values are not too reliable as they're updated only when the `onChainChanged` event fires.
      *      And this doesn't happen when the connection is initiated.
      *      For WalletConnect we could do that manually as we control what chain we connect to.
      *      For BrowserWallet we don't have that option (see also https://concordium.atlassian.net/browse/CBW-633).
      */
-    activeConnectionGenesisHash: string | undefined;
+    genesisHashes: Map<WalletConnection, string>;
+}
 
-    /**
-     * The selected account of {@link activeConnection}.
-     */
-    activeConnectedAccount: string | undefined;
+function updateMapEntry<K, V>(map: Map<K, V>, key: K | undefined, value: V | undefined) {
+    const res = new Map(map);
+    if (key) {
+        if (value) {
```

Comment:
```suggestion
        if (value !== undefined) {
```
Right?

Reply:
It should never be called with falsy values that aren't undefined, but I guess it's fine to add this. Will also add it in the key check.